ipc/cli/src/commands/wallet/balances.rs

```diff
@@ -50,16 +49,19 @@ impl CommandLineHandler for WalletBalances {
 
                 let v: Vec<anyhow::Result<(TokenAmount, &EthKeyAddress)>> = join_all(r).await;
 
-                for r in v.into_iter().filter_map(|r| r.ok()) {
-                    let (balance, addr) = r;
-                    if addr.to_string() != "default-key" {
-                        println!("{} - Balance: {}", addr.to_string(), balance);
-                        no_balance.retain(|a| a != addr);
+                for r in v.into_iter() {
+                    match r {
+                        Ok(i) => {
+                            let (balance, addr) = i;
+                            if addr.to_string() != "default-key" {
+                                println!("{} - Balance: {}", addr.to_string(), balance);
+                            }
+                        }
+                        Err(e) => {
+                            return Err(e);
```

Comment:
Okay so we went from ignoring errors with `.filter_map(|r| r.ok())` and printing 0 balance for them to printing the first few successful balances until we encounter an error, where we stop going down the list and return an overall failure. 

That seems a bit inconsistent, that depending on which of the N requests failed I see more or less results, compared to for example:
* Print all results if all of them succeeded or none if any of them failed
* Print all results as JSON regardless of error or failure, but if it failed then print eg.. `{addr: "...", error: "oh noes!"}`
* Print all the successful results first and then return an overall error if any failed

The all-or-nothing strategy can be achieved for example like so:

```rust
let v: Vec<(TokenAmount, &EthKeyAddress)> = join_all(r).await.collect::<anyhow::Result<Vec<_>>>()?;
```

Reply:
This one is a bit tricky, because there are errors for which all balances will fail with the same error (like RPC unavailable), and others that may be address-specific (like address not initialized on-chain). I agree that my approach may have not been the best because it was non-deterministic, so I decided to go with your suggestion of printing all of the balances that succeed, and then returning the errors (if any) at the end. That way if it this is a balance-wide error (like RPC) we return a single error, and if it is address-specific we print the balances and then the errors. Let me know what you think.